src/libs/Navigation/CustomActions.js

```diff
@@ -42,55 +43,94 @@ function navigateBackToRootDrawer(navigationRef) {
+        const newScreenName = getScreenNameFromState(newStateFromRoute);
+        const newScreenParams = getParamsFromState(newStateFromRoute);
 
-        // Avoid the navigation and refocus the report if we're trying to navigate to our active report
-        // We use our RootState as the dispatch's state is relative to the active navigator and might
-        // not contain our active report.
-        const rootState = navigationRef.current.getRootState();
-        const activeReportID = lodashGet(rootState, 'routes[0].state.routes[0].params.reportID', '');
-        if (state.type !== 'drawer') {
-            navigateBackToRootDrawer(navigationRef);
+        // When we are navigating away from a non-drawer navigator we need to first dismiss any screens pushed onto the main stack.
+        if (currentState.type !== 'drawer') {
+            navigateBackToRootDrawer();
         }
-        if (activeReportID === params.reportID) {
+
+        // If we're trying to navigate to the same screen that is already active there's nothing more to do except close the drawer.
+        // This prevents unnecessary re-rendering the screen and adding duplicate items to the browser history.
+        const activeState = getActiveState();
+        const activeScreenName = getScreenNameFromState(activeState);
+        const activeScreenParams = getParamsFromState(activeState);
+        if (newScreenName === activeScreenName && _.isEqual(activeScreenParams, newScreenParams)) {
             return DrawerActions.closeDrawer();
         }
 
-        // When navigating from non Drawer navigator, get new state for the report and reset the navigation state.
-        if (state.type !== 'drawer') {
-            state = linkingConfig.getStateFromPath
-                ? linkingConfig.getStateFromPath(path, linkingConfig.config)
-                : getStateFromPath(path, linkingConfig.config);
+        let state = currentState;
+
+        // When navigating from non-Drawer navigator we switch to using the new state generated from the provided route. If we are navigating away from a non-Drawer navigator the
+        // currentState will not have a history field to use. By using the state from the route we create a "fresh state" that we can use to setup the browser history again.
+        // Note: A current limitation with this is that navigating "back" won't display the routes we have cleared out e.g. SearchPage and the history effectively gets "reset".
+        if (currentState.type !== 'drawer') {
+            state = newStateFromRoute;
         }
 
-        const screenRoute = {type: 'route', name: screenName};
-        const history = _.map([...(state.history || [screenRoute])], () => screenRoute);
+        const screenRoute = {type: 'route', name: newScreenName};
+        const history = _.map(state.history ? [...state.history] : [screenRoute], () => screenRoute);
 
-        // Force drawer to close and show the report screen
+        // Force drawer to close and show
```

Comment:
NAB: I think this is removed mistakenly or `and show` can be removed as well.

```suggestion
        // Force drawer to close
```

Reply:
Ah yea, not a mistake because I didn't want to have the "report page" referenced here.

Reply:
Oh I see what you mean actually. It does kind of seem like an unfinished thought... "and show... what?"